### PR TITLE
CFD-285 : Homepage : Voices & views carousel.

### PR DESCRIPTION
**Pull request by nader77**
#468

### CODE REVIEW

**Comment by nader77** [comment]
Note: On Initial load sometimes the carousel gets stuck.

**Comment by nader77** [comment]
Update: Initial load issue fixed.
This PR covers the following:

* Add carousel to the anonymous homepage, This carousel is mobile friendly, Works with finger swipe.
* The carousel fetches up to 6 articles (V&V) which must have Intro text and banner field filled.
* The V&V are ordered by sticky first, promoted to homepage second, Creation time after that.
* Add some fields in the dummy migration so we can have the carousel already showing on the homepage.

**Comment by nader77** [comment]
Carousel with dummy content:
carousel

**Comment by nader77** [comment]
@HelenaEksler PR is ready for code review.

**Comment by nader77** [comment]
Also in this PR:
* Removed some unneeded AngularJs dependencies.
* Changed the port for live reload, so it's possible to load the theme and the app together.
* Removed the debugger for the app. (Not needed anymore).

File: capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module [hunk 1]
```diff
@@ -5,3 +5,76 @@
...
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->fieldCondition('c4m_banner', 'fid', '', '!=')
+    ->fieldCondition('c4m_intro_text', 'value', '', '!=')
+    ->propertyOrderBy('sticky', 'DESC')
+    ->propertyOrderBy('promote', 'DESC')
+    ->propertyOrderBy('created', 'DESC')
+    ->range(0, 6)
+    ->execute();
+
+  if (!empty($result['node'])) {
```
Comment (HelenaEksler) [inline capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module]: Return early.
```php
if (empty($result['node'])) {
  return;
}``

File: capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module [hunk 2]
```diff
@@ -5,3 +5,76 @@
...
+  $result = $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'article')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->fieldCondition('c4m_banner', 'fid', '', '!=')
+    ->fieldCondition('c4m_intro_text', 'value', '', '!=')
+    ->propertyOrderBy('sticky', 'DESC')
+    ->propertyOrderBy('promote', 'DESC')
+    ->propertyOrderBy('created', 'DESC')
+    ->range(0, 6)
```
Comment (HelenaEksler) [inline capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module]: Shouldn't it be 3?

**Comment by HelenaEksler** [comment]
@nader77 Due to wiki I should get the first one after the third one (has an infinite loup (1->2, 2->3, 3->1, ...)). Now I have nothing after the third one and before the 1-st one.

**Comment by HelenaEksler** [comment]
Add this block also to the authenticated user variant.

**Comment by nader77** [comment]
@HelenaEksler The directive i used does not support the loop, need to look for another directive.

**Comment by nader77** [comment]
I will add it when we merge "My groups" branch, because it fixes the homepage layout.

**Comment by nader77** [comment]
@HelenaEksler Pleaser re-install and review :)

**Comment by nader77** [comment]
@HelenaEksler Sorry, Don't review, we will keep this open until other branches are merged and we can complete this one.
Thanks.

**Comment by HelenaEksler** [comment]
@zero2one re-rolled.

**Pull request merged**